akka-typed-tests/src/test/scala/akka/typed/persistence/scaladsl/PersistentActorCompileOnlyTest.scala

```diff
@@ -372,15 +373,15 @@ object PersistentActorCompileOnlyTest {
         cmd match {
           case Greet(whom) ⇒
             println(s"Hi there, I'm $state!")
-            Effect.done
+            Effect.none
           case CheerUp(sender) ⇒
             changeMoodIfNeeded(state, Happy)
               .andThen { sender ! Ack }
           case Remember(memory) ⇒
             // A more elaborate example to show we still have full control over the effects
             // if needed (e.g. when some logic is factored out but you want to add more effects)
             val commonEffects = changeMoodIfNeeded(state, Happy)
-            Effect.persistAll(commonEffects.events :+ Remembered(memory), commonEffects.sideEffects)
+            Effect.persist(commonEffects.events :+ Remembered(memory), commonEffects.sideEffects)
```

Comment:
I'm wondering if we could improve on this and make it a little bit more convenient. Something like:

```scala
commandEffects
   .append(Remembered(memory))
   .andThen(...)
```

I believe the only reason for having a `Seq` of SideEffects to be able to reuse them like in the example above. But in general, we are creating a linked list of chainable effects that are later executed in order. 

If we add `append` and `prepend` methods on the `Effect`, we can have one SideEffect, instead of Seq.

That's just some thoughts. Maybe topic for discussion and eventually a new PR.

Reply:
In this example, would the side effects in `commandEffects` be executed before or after persisting `Remembered(memory)`?

Reply:
The proposal does not sounds readable to me, please open separate ticket if we need to discuss it

Reply:
Always after, but if you are asking is because the API is not clear on that. So, maybe not a good idea.

The result of the `append` or `prepend` is a new `Effect` with an augmented list of events and the previously registered side-effects. On that new `Effect`, we could add another side-effect and so on.